### ISSUE TITLE
[Telemetry] Dashboard for LocalNet logs viewing & filtering

### ISSUE THREAD

**Issue by okdas** [pokt-network/pocket#189]
## Objective

Build a Grafana dashboard to aggregate, view, and filter logs. This is blocked by:
1. https://github.com/pokt-network/pocket/issues/181
2. https://github.com/pokt-network/pocket/issues/164

## Goals

- [x] Filter logs by source module 
- [x] Filter logs by log level
- [ ] Enable pre-configured filters

## Deliverable

- [x] Updates to the grafana dashboard to allow filtering all logs by source module
- [x] Updates to the grafana dashboard to allow filtering all logs by filter
- [ ] Pre-configure filters
- [ ] Updated documentation to show how to use / enable this

## Non-goals / Non-deliverables

- Though these dashboards might be reused in DevNets, compitabilty is not the goal

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @okdas 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
@okdas Please delete these moving forward when you create new issues:

<img width="869" alt="Screen Shot 2022-09-04 at 3 25 12 PM" src="https://user-images.githubusercontent.com/1892194/188330260-84369bd2-f22c-4275-8dc1-c6ad98d7aa5a.png">

**Comment by jessicadaugherty** [comment]
Hey @okdas! A few questions about this ticket:

1. Where are logs being aggregated (where does this dashboard live)?
2. Are the logs normalized or will they need to be as part of this effort?
3. What will trigger the updates (i.e. cron job)?
4. May depend on the answer to question #1, but what does filtering mean here (i.e. a UI or a query)?

**Comment by jessicadaugherty** [comment]
Looks like the answer to #2 is yes, they are standardized per https://github.com/orgs/pokt-network/projects/142/views/12?filterQuery=log

And also looks like we are using Grafana?

**Comment by okdas** [comment]
@jessicadaugherty 

> Where are logs being aggregated (where does this dashboard live)? 

This effort is to build dashboards for local development - we already have some, but need to do more work there.
Local development instructions are available [here](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md).

> Are the logs normalized or will they need to be as part of this effort?

We have some tickets to have a better logging strategy in place! :) Currently logs are not structured well and are hard to parse. We should utilize metrics whenever possible though and only resort to parsing logs when it's necessary.

> What will trigger the updates (i.e. cron job)?

I am a bit confused about updates - we don't really need to trigger any updates, as data is readily available in realtime from the databases.

> what does filtering mean here (i.e. a UI or a query)?

Filtering data with logging queries - correct! We are utilizing Loki for localnet, and I'm almost 100% sure that is also going to be the case for dashboards for DevNet. So query language we are going to use for logs is [LogQL](https://grafana.com/docs/loki/latest/logql/).


Let's talk about that stuff more on our 1:1. There might be some confusion about how tasks are organized and I'm happy to change/modify the tasks so they make more sense.

**Comment by Olshansk** [comment]
I think what would help is being explicit about the exact goals/deliverables of this ticket.

This is blocked by:
1. https://github.com/pokt-network/pocket/issues/181
2. https://github.com/pokt-network/pocket/issues/164

Once those two tasks are done, the goals this ticket are:
- [ ] Filter logs by source module
- [ ] Filter logs by log level

The deliverables would include:
- [ ] Updates to the grafana dashboard to allow filtering all logs by source module
- [ ] Updates to the grafana dashboard to allow filtering all logs by filter
- [ ] Updated documentation to show how to use / enable this

@okdas Lmk if I understood the scope correctly

**Comment by jessicadaugherty** [comment]
Exactly, thanks @Olshansk! Will review with @okdas and get this back into the flow

**Comment by jessicadaugherty** [comment]
May be fixed by Logging PR

**Comment by Olshansk** [comment]
@okdas Could you create a new ticket to track the issue around the attributes/labels we have remaining? We can then close this one out and have the follow-up work as a separate ticket

**Comment by okdas** [comment]
We have a #600 to address this.
